src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -148,12 +167,12 @@ def add_cloud(cloud):
         config.add_section(cloud.name)
     except configparser.DuplicateSectionError:
         raise CloudAlreadyRegisteredException(cloud.name)
-    for endpoint in cloud.endpoints:
-        if cloud.endpoints[endpoint]:
-            config.set(cloud.name, 'endpoint_{}'.format(endpoint), cloud.endpoints[endpoint])
-    for suffix in cloud.suffixes:
-        if cloud.suffixes[suffix]:
-            config.set(cloud.name, 'suffix_{}'.format(suffix), cloud.suffixes[suffix])
+    for k, v in cloud.endpoints.__dict__.items():
+        if v:
```

Comment:
I don't want to save the values that weren't set by the user to the cloud config file.
e.g. If my cloud doesn't support Data Lake, there's no need to set the Data Lake endpoint.